### PR TITLE
Sleeping carp no longer deflects validhunter bullets

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Sleeping carp no longer deflects validhunter bullets

## Why It's Good For The Game

You can validtest people with carp by shooting them with invisible revolver, laser tag bullets and foam darts.

## Changelog
:cl:
tweak: sleeping carp no longer deflects harmless projectiles.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/special/wormhole.dm [hunk 1]
```diff
@@ -11,6 +11,7 @@
 	impact_type = /obj/effect/projectile/impact/wormhole
 	muzzle_type = /obj/effect/projectile/muzzle/wormhole
 	hitscan = TRUE
+	martial_arts_no_deflect = FALSE
```
Comment (PowerfulBacon) [inline code/modules/projectiles/projectile/special/wormhole.dm]:
```suggestion
	martial_arts_no_deflect = TRUE
```

**Comment by Cenrus** [comment]
Some other issues, finger gun projectiles can waste charges of shielded suits (Think cultist robes, nuke op shielded variant and wizard battlearmor).

File: code/modules/projectiles/projectile/magic.dm [hunk 2]
```diff
@@ -375,6 +383,11 @@
 	icon_state = "locker"
 	nodamage = TRUE
 	flag = "magic"
+<<<<<<< Updated upstream
+=======
+	martial_arts_no_deflect = FALSE
+	var/weld = TRUE
+>>>>>>> Stashed changes
```
Comment (PowerfulBacon) [inline code/modules/projectiles/projectile/magic.dm]:
```suggestion
	martial_arts_no_deflect = FALSE
	var/weld = TRUE
```

**Comment by bloons3** [comment]
Guess I'll have to shoot everyone with real bullets then, what a killjoy /s

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 3]
```diff
@@ -759,15 +759,20 @@
 		s.set_up(2, 1, src)
 		s.start()
 		owner.visible_message("<span class='danger'>[owner]'s shields deflect [attack_text] in a shower of sparks!</span>")
+		if(isprojectile(hitby))
+			var/obj/item/projectile/P = hitby
+			//Fake bullets just disintegrate
+			if(hitby.martial_arts_no_deflect)
```
Comment (PowerfulBacon) [inline code/modules/clothing/spacesuits/hardsuit.dm]:
```suggestion
			if(P.martial_arts_no_deflect)
```

**Comment by super12pl** [comment]
While I get that validtesting carp users using fingerguns and such is bad and not intended...
sleeping carp really doesn't have real downsides outside of this

**Comment by ToasterBan** [comment]
> sleeping carp really doesn't have real downsides outside of this

can't use ranged weapons

**Comment by yyzsong** [comment]
It costs like 16 tc lol

**Comment by Effemu** [comment]
it still deflects riots right?

**Review by utilbird — APPROVED** [review]
You guys don't antag test with a .50 cal?

**Comment by ghost** [comment]
I was so fucking worried about this PR nerfing carp. Bacon is the most based of coders

**Comment by jupyterkat** [comment]
https://images-ext-1.discordapp.net/external/-6RcqozKvLpNTG1zmkypEcIUsTuytHTk3lAfcefe43c/https/media.discordapp.net/attachments/689018783712411682/755167679261835374/image0.jpg

**Comment by jupyterkat** [comment]
image0

**Comment by PowerfulBacon** [comment]
> 
> 
> it still deflects riots right?

Yes

**Comment by PowerfulBacon** [comment]
> 
> 
> image0

Seems to be the current beaviour

**Comment by 123chess456** [comment]
> While I get that validtesting carp users using fingerguns and such is bad and not intended...
> sleeping carp really doesn't have real downsides outside of this

This is mainly because security doesn't take advantage of the carp counters available to them. For example, flashes, teargas, pepper spray, or the pretty amazing energy bola. Even building securitrons can work wonders.

**Comment by bloons3** [comment]
> flashes, teargas, pepper spray

All 3 of these are bypassed with sunglasses, a face mask, and earmuffs or a bowman headset. All of which can be found in any Security locker.

**Comment by 123chess456** [comment]
> > flashes, teargas, pepper spray
> 
> All 3 of these are bypassed with sunglasses, a face mask, and earmuffs or a bowman headset. All of which can be found in any Security locker.

That's why bolas and DRAGnets exist and can be taken roundstart at sec. Heck even print a bear trap. You apply the slow then you can stunbaton sleeping carp dude into submission if you're robust. Get a riot shield if you wanna be extra careful.

If you're willing to do some legwork as sec you can also get the CMO's hypospray and fill it with pax to remove their ability to attack you or make robotics create securitrons. Securitrons meme on sleeping carp users since they melee attack em into submission.

There's a lot of ways to counter sleeping carp dudes but nobody seems to use them.

**Comment by PowerfulBacon** [comment]
sleeping carp is pretty powerful (albeit very easy to stop with 2 seccies together with a stun baton assuming they don't have EMPs) however still shouldn't have a way to cheese with bullets that don't exist like the fingergun rounds.

**Pull request merged**